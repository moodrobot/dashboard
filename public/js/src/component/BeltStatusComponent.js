import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import renderHTML from "react-render-html";

class BeltStatusComponent extends PureComponent {
  constructor(props) {
    super(props);
  }

  nextBeltPoints() {
    let next_belt_points = null;

    switch (this.props.belt) {
      case "":
        next_belt_points = 50;
        break;
      case "green":
        next_belt_points = 500;
        break;
      case "brown":
        next_belt_points = 5000;
        break;
      default:
        next_belt_points = 0;
    }

    return next_belt_points;
  }

  render() {
    //console.log("RENDERED BELT COMPONENT");
    const T = this.props.translate;

    return (
      <div className="row no-gutters">
        <div className="col-md-3 col-lg-2">
          {this.props.belt != "" && <span className={"icon-belt " + this.props.belt} />}
        </div>
        <div className="col-md-9 col-lg-10 vtop">
          <p className="title">
            {T["status-" + this.props.belt]}
            <br />
            {renderHTML(T["information"])}
          </p>

          <p className="title">{T["points-activity"]}</p>
          <p>
            <span>{T["status-title"]}</span>{" "}
            {parseInt(this.props.points, 10).toLocaleString(this.props.currentLanguage)}
          </p>
          <p>
            <span>{T["total-title"]}</span>{" "}
            {parseInt(this.props.total_points, 10).toLocaleString(this.props.currentLanguage)}
          </p>
          <p>
            {this.nextBeltPoints() > 0 && (
              <span>
                {T["next-belt-level-" + this.props.belt + "-title"]}
                {" " + parseInt(this.nextBeltPoints(), 10).toLocaleString(this.props.currentLanguage)}
                {" " + T["next-belt-level-points"]}
              </span>
            )}
          </p>
        </div>
      </div>
    );
  }
}

BeltStatusComponent.propTypes = {
  translate: PropTypes.object,
  currentLanguage: PropTypes.string,
  belt: PropTypes.string,
  points: PropTypes.string,
  total_points: PropTypes.string
};

export default BeltStatusComponent;
