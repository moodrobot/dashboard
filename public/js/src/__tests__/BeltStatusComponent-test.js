import React from "react";
import { shallow } from "enzyme";
import BeltStatusComponent from "../component/BeltStatusComponent";
import i18n from "../i18n-for-testing"; // initialized i18next instance for testing

describe("<BeltStatusComponent />", () => {
  it("expect to match snapshot", () => {
    const wrapper = shallow(
      <BeltStatusComponent
        belt="blue"
        translate={i18n.t("profile-details-belt", { returnObjects: true })}
        currentLanguage={"en-US"}
        points="9000"
        total_points="9001"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
