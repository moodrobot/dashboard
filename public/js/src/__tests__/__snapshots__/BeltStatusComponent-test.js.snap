// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BeltStatusComponent /> expect to match snapshot 1`] = `
<div
  className="row no-gutters"
>
  <div
    className="col-md-3 col-lg-2"
  >
    <span
      className="icon-belt blue"
    />
  </div>
  <div
    className="col-md-9 col-lg-10 vtop"
  >
    <p
      className="title"
    >
      You are a blue belt
      <br />
      <a
        href="https://software.intel.com/en-us/blackbelt"
        key="0"
      >
        Learn about belt status &gt;
      </a>
    </p>
    <p
      className="title"
    >
      Points Activity
    </p>
    <p>
      <span>
        Current Status Points:
      </span>
       
      9,000
    </p>
    <p>
      <span>
        Total Lifetime Points:
      </span>
       
      9,001
    </p>
    <p />
  </div>
</div>
`;
